aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -344,6 +344,25 @@ mod loader {
             self
         }
 
+        /// Override the name of the app used to build [`SdkConfig`](aws_types::SdkConfig).
+        ///
+        /// This _optional_ name is used to identify the application in the user agent that
+        /// gets sent along with requests.
+        ///
+        /// # Examples
+        /// ```no_run
+        /// # async fn create_config() {
+        /// use aws_config::AppName;
+        /// let config = aws_config::from_env()
+        ///     .app_name(AppName::new("my-app-name").expect("valid app name"))
+        ///     .load().await;
+        /// # }
+        /// ```
+        pub fn app_name(mut self, app_name: AppName) -> Self {
+            self.app_name = Some(app_name);
```

Comment:
Do we need to add any sort of validation for this being non-empty string or is that handled by `AppName`?

Reply:
Thanks for pointing it out! It's good to guard against that. Addressed in b296de9.